File: backend/backend.Domain/src/Entities/OrderProduct.cs
namespace backend.Domain.src.Entities
{
    public class OrderProduct : BaseEntity
    {
        public Guid OrderId { get; set; }
        public Guid ProductId { get; set; }
        public int Quantity { get; set; }
        public string? Color { get; set; }
        public Order Order { get; set; }
        public Product Product { get; set; }
        
    }
}

File: frontend/src/types/OrderProduct.ts
import { Guid } from "guid-typescript"
import Product from "./Product"
import OrderProductShipmentRead from "./Shipment" 

export interface OrderProductCreate
{
    orderId: Guid
    productId: Guid
    quantity: number
    color?: string
}

export interface OrderProductRead
{
    id: Guid
    orderId: Guid
    productId: Guid
    quantity: number
    color?: string
}

export interface OrderOfOrderProductRead
{
    product: Product
    quantity: number
    color?: string
    shipment: OrderProductShipmentRead
}

